app/controllers/bookmarks_controller.rb

```diff
@@ -74,4 +74,29 @@ def toggle_pin
 
     render json: failed_json.merge(errors: bookmark_manager.errors.full_messages), status: 400
   end
+
+  def bulk
+    if params[:bookmark_ids].present?
+      unless Array === params[:bookmark_ids]
+        raise Discourse::InvalidParameters.new(
+                "Expecting bookmark_ids to contain a list of bookmark ids",
+              )
+      end
+      bookmark_ids = params[:bookmark_ids].map { |t| t.to_i }
+    else
+      raise ActionController::ParameterMissing.new(:bookmark_ids)
+    end
+
+    operation =
+      params
+        .require(:operation)
+        .permit(:type, *DiscoursePluginRegistry.permitted_bulk_action_parameters)
```

Comment:
I think we can keep the bookmark operation types separate from the topic bulk action operation types here.